dev_scripts_helpers/thin_client/test/test_setenv.py

```diff
@@ -0,0 +1,231 @@
+import logging
+import os
+import pathlib
+from typing import Tuple
+
+import pytest
+
+import helpers.hdbg as hdbg
+import helpers.hgit as hgit
+import helpers.hio as hio
+import helpers.hsystem as hsystem
+import helpers.hunit_test as hunitest
+
+_LOG = logging.getLogger(__name__)
+
+
+# #############################################################################
+# _SetenvTestCase
+# #############################################################################
+
+
+class _SetenvTestCase(hunitest.TestCase):
+    """
+    Helper test class to perform common setup and provide utility methods for
+    testing setenv.sh script.
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup_teardown_test(self) -> None:
+        self.set_up_test()
+
+    def set_up_test(self) -> None:
+        """
+        Set up test environment.
+        """
+        # Get the path to the `setenv.sh` script.
+        git_root = hgit.find_git_root()
```

Comment:
Can you add a TODO here saying that maybe we should factor out these functions in one of the helpers? E.g., henv.py (although there might be some import dep issues).

E.g., "get sentenv", "get repo_config", ... seem general functionalities.